### PR TITLE
Added string Concatenation

### PR DESCRIPTION
**Pull request by ansi22-zz**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

(Yes)

### Description

Added string concatenation problem that solves issue #3362 

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

(#3362 )

### CODE REVIEW

File: C-Plus-Plus/other/n_string_concatenate.cpp [hunk 1]
```diff
@@ -0,0 +1,43 @@
+/*
+String Concatenation 
+Problem Statement : Take n string from user and concatenate them together.
+Input Format : First line takes number of strings to be concatenated.
+               Next n lines contains the strings to be concatenated.
+Output Format : Print a single line that contains the concatenated string.
+Note : No space is there in between a single string.
+Sample Test Case :
```
Comment (rohanbabbar04) [inline C-Plus-Plus/other/n_string_concatenate.cpp]: Add the sample test case after the code

File: C-Plus-Plus/other/n_string_concatenate.cpp [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+           Take a string variable and update it using append() function each time a new sting is entered.
+           Print the string.
+*/
+
+#include <iostream>
+#include <string>
+using namespace std;
+
+int main() {
+    //n stores the no. of string
```
Comment (rohanbabbar04) [inline C-Plus-Plus/other/n_string_concatenate.cpp:25]: No need to explain each line just the important ones

File: C-Plus-Plus/other/n_string_concatenate.cpp [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+	int n; 
+    //takes inut from the user
```
Comment (rohanbabbar04) [inline C-Plus-Plus/other/n_string_concatenate.cpp]: No need to explain each line

File: C-Plus-Plus/other/n_string_concatenate.cpp [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+	cin>>n; 
+    //f_str contains the final string  
+	string f_str="";
+    //loop for taking n strings from user and updating the final string 
+	for(int i=0;i<n;i++)
```
Comment (rohanbabbar04) [inline C-Plus-Plus/other/n_string_concatenate.cpp]: Improper indentation

File: C-Plus-Plus/other/n_string_concatenate.cpp [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+    {
+    //variable to take string as input
+    string a;
+    cin>>a;
+    //f_str.append() appends the string each time a new string is entered 
+    f_str.append(a);
+    }
```
Comment (rohanbabbar04) [inline C-Plus-Plus/other/n_string_concatenate.cpp:39]: Use a C++ formatter  to format your code

File: C-Plus-Plus/other/n_string_concatenate.cpp [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+    //prints the final string
+    cout<<f_str;
+	return 0;
+}
```
Comment (rohanbabbar04) [inline C-Plus-Plus/other/n_string_concatenate.cpp]: Add Time Complexity and Space Complexity 
Add a new line at EOF

**Review by rohanbabbar04 — CHANGES_REQUESTED** [review]

**Comment by ansi22-zz** [comment]
Hello mentors,
I have made the requested changes and updated the code.
Do check it and kindly let me know if more changes are required.
@HarshCasper @ankitaggarwal23 @bhav09

File: C-Plus-Plus/other/n_string_concatenate.cpp [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+    cin>>n; 
+    //f_str contains the final string  
+    string f_str="";
+    //loop for taking n strings from user and updating the final string 
+    for(int i=0;i<n;i++)
+    {
+    string a;
+    cin>>a;
+    //f_str.append() appends the string each time a new string is entered 
+    f_str.append(a);
```
Comment (rohanbabbar04) [inline C-Plus-Plus/other/n_string_concatenate.cpp]: Improper Indentation

**Review by rohanbabbar04 — CHANGES_REQUESTED** [review]

File: C-Plus-Plus/other/n_string_concatenate.cpp [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+Output Format : Print a single line that contains the concatenated string.
+Note : No space is there in between a single string.
+
+Approach : Take n strings as input using loop.
+           Take a string variable and update it using append() function each time a new sting is entered.
+           Print the string.
+*/
+
+#include <iostream>
+#include <string>
```
Comment (rohanbabbar04) [inline C-Plus-Plus/other/n_string_concatenate.cpp]: Instead of this use #include<bits/stdc++.h> which includes all the above libraries

**Review by rohanbabbar04 — CHANGES_REQUESTED** [review]

File: C-Plus-Plus/other/n_string_concatenate.cpp [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+
+Approach : Take n strings as input using loop.
+           Take a string variable and update it using append() function each time a new sting is entered.
+           Print the string.
+*/
+
+#include <bits/stdc++.h>
+using namespace std;
+
+int main() {
```
Comment (HarshCasper) [inline C-Plus-Plus/other/n_string_concatenate.cpp]: Don't push the main logic of the code in the main() function. Use main() only for the driver code

**Review by HarshCasper — CHANGES_REQUESTED** [review]

**Review by rohanbabbar04 — APPROVED** [review]

File: C-Plus-Plus/other/n_string_concatenate.cpp [hunk 10]
```diff
@@ -0,0 +1,58 @@
...
+
+//within concat_str(...) final string is calculated
+void concat_str(vector<string> str,int size)
+{
+    string f_str="";
+    for(int i=0;i<size;i++)
+    {   //append(...) adds each string into f_str
+        f_str.append(str[i]);
+    }
+    //final output
```
Comment (HarshCasper) [inline C-Plus-Plus/other/n_string_concatenate.cpp]: Redundant comment

File: C-Plus-Plus/other/n_string_concatenate.cpp [hunk 11]
```diff
@@ -0,0 +1,58 @@
...
+
+Sample Test Case :
+Input : 4
+        "This"
+        "is"
+        "Concatenated"
+        "String."
+Output : "ThisisConcatenatedString."
+*/
+
```
Comment (HarshCasper) [inline C-Plus-Plus/other/n_string_concatenate.cpp]: Remove this line of code

**Review by HarshCasper — CHANGES_REQUESTED** [review]

**Review by HarshCasper — APPROVED** [review]

**Pull request merged**